rust/lance-file/src/v2/writer.rs

```diff
@@ -802,4 +824,150 @@ mod tests {
+
+        // almost 8MB
+        let data: Vec<u64> = (0..1_000_000).collect();
+        let array = UInt64Array::from(data);
+        let batch =
+            RecordBatch::try_new(arrow_schema.clone().into(), vec![Arc::new(array)]).unwrap();
+
+        let options = FileWriterOptions {
+            max_page_bytes: Some(1024 * 1024), // 1MB
+            ..Default::default()
+        };
+
+        let tmp_dir = tempdir().unwrap();
+        let path = tmp_dir.path().join("test.lance");
+        let object_store = ObjectStore::local();
+        let mut writer = FileWriter::try_new(
+            object_store
+                .create(&Path::from(path.to_str().unwrap()))
+                .await
+                .unwrap(),
+            lance_schema,
+            options,
+        )
+        .unwrap();
+
+        writer.write_batch(&batch).await.unwrap();
+        writer.finish().await.unwrap();
+
+        let fs = FsFixture::default();
+        let file_scheduler = fs
+            .scheduler
+            .open_file(
+                &Path::from(path.to_str().unwrap()),
+                &CachedFileSize::unknown(),
+            )
+            .await
+            .unwrap();
+        let file_reader = FileReader::try_open(
+            file_scheduler,
+            None,
+            Arc::<DecoderPlugins>::default(),
+            &FileMetadataCache::no_cache(),
+            FileReaderOptions::default(),
+        )
+        .await
+        .unwrap();
+
+        let column_meta = file_reader.metadata();
+
+        let mut total_page_num: u32 = 0;
+        for (col_idx, col_metadata) in column_meta.column_metadatas.iter().enumerate() {
+            assert!(
+                !col_metadata.pages.is_empty(),
+                "Column {} has no pages",
+                col_idx
+            );
+
+            for (page_idx, page) in col_metadata.pages.iter().enumerate() {
+                total_page_num += 1;
+                let total_size: u64 = page.buffer_sizes.iter().sum();
+                assert!(
+                    total_size <= 1024 * 1024,
+                    "Column {} Page {} size {} exceeds 1MB limit",
+                    col_idx,
+                    page_idx,
+                    total_size
+                );
+            }
+        }
+
+        assert_eq!(total_page_num, 8)
+    }
+
+    #[tokio::test(flavor = "current_thread")]
+    async fn test_max_page_bytes_env_var() {
+        let arrow_field = Field::new("data", DataType::UInt64, false);
+        let arrow_schema = Schema::new(vec![arrow_field]);
+        let lance_schema = LanceSchema::try_from(&arrow_schema).unwrap();
+        // almost 4MB
+        let data: Vec<u64> = (0..500_000).collect();
+        let array = UInt64Array::from(data);
+        let batch =
+            RecordBatch::try_new(arrow_schema.clone().into(), vec![Arc::new(array)]).unwrap();
+
+        std::env::set_var(ENV_LANCE_FILE_WRITER_MAX_PAGE_BYTES, "2097152");
```

Comment:
Our CI runs tests in parallel so setting environment variables in unit tests is always a little tricky.  However, I'm not aware of any other tests that rely on a particular max page size (if they use file reader they don't test page size and if they do test max page size they are at a lower level or setting the variable directly) so I think we will get away with it.